subiquity/common/apidef.py

```diff
@@ -230,7 +230,7 @@ class guided:
             def GET(wait: bool = False) -> GuidedStorageResponse:
                 pass
 
-            def POST(choice: Optional[GuidedChoice]) \
+            def POST(data: Payload[GuidedChoice]) \
```

Comment:
Are we sure that the other API consumers (e.g. desktop installer) never pass `None` / `null` to this endpoint?
Otherwise, we might need to keep `Payload[Optional[GuidedChoice]]` and the check for `choice is not None` on the server side.